optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -490,7 +515,8 @@ def __init__(self, session: ort.InferenceSession, parent_model: ORTModel):
         self.output_names = {output_key.name: idx for idx, output_key in enumerate(self.session.get_outputs())}
         config_path = Path(session._model_path).parent / self.CONFIG_NAME
         self.config = self.parent_model._dict_from_json_file(config_path) if config_path.is_file() else {}
-        self.input_dtype = {inputs.name: _ORT_TO_NP_TYPE[inputs.type] for inputs in self.session.get_inputs()}
+        self.input_dtypes = {inputs.name: inputs.type for inputs in self.session.get_inputs()}
```

Comment:
all pipelines needs to be updated no https://github.com/huggingface/optimum/blob/5ece6e88156de6a0de848ad6c1af381c81cfbcc3/optimum/pipelines/diffusers/pipeline_stable_diffusion_xl.py#L107C66-L107C77

Reply:
not sure it's worth it but if we want to change the attribute name we should have something like that for a couple of release :
```python
 @property
  def input_dtype(self):
        # add warning that this will be deprecated in optimum vX.X.X
        return self.input_dtypes
```

Reply:
The `dtype` for diffusion is not really needed here and generally not used (should be tbh, to cast input ids to the right dtype), I changed it here to conform with the base model part (in base.py).
I'll revert it for now and we can add that later on.